client/components/MainSettings.jsx

```diff
@@ -19,45 +20,62 @@ const MainSettings = React.createClass({
     });
   },
 
-  componentWillUnmount() {
-    Meteor.call("room.archive", this.props.room._id, this.state.archived);
-  },
-
   toggleArchivedState() {
-    Meteor.call("room.archive", this.props.room._id, !this.state.archived);
+    const params = {
+      archived: !this.state.archived,
+      roomId: this.props.room._id
+    };
+    Meteor.call("room.archive", params);
     this.setState({archived: !this.state.archived});
```

Comment:
Made the changes, no longer using state to keep track of archived field. Also moved the archived field to userRoom model, where it should have been from start.